src/core/shape_inference/include/fft_base_shape_inference.hpp

```diff
@@ -0,0 +1,133 @@
+        const auto& signal_size_shape = input_shapes[2];
+        NODE_VALIDATION_CHECK(op,
+                              signal_size_shape.rank().compatible(1),
+                              "FFT op signal size input must be 1D tensor. Got signal: ",
+                              signal_size_shape);
+
+        if (axes_shape.is_static() && signal_size_shape.is_static()) {
+            NODE_VALIDATION_CHECK(op,
+                                  axes_shape[0].compatible(signal_size_shape[0]),
+                                  "Sizes of inputs 'axes' and 'signal_size' must be equal. Got "
+                                  "size of 'axes': ",
+                                  axes_shape[0],
+                                  "size of 'signal_size': ",
+                                  signal_size_shape[0]);
+        }
+    }
+
+    output_shape = input_shape;
+    if (input_shape.rank().is_static() && axes_shape.rank().is_static() && input_shapes.size() == 3 && axes_are_known) {
+        const auto& signal_size_shape = input_shapes[2];
+        std::vector<int64_t> signal_size;
+        bool status_signal_size = get_data_as_int64<T>(2, op, signal_size, constant_data);
+
+        if (signal_size_shape.rank().is_static() && status_signal_size) {
+            size_t num_of_axes = axes.size();
+            for (size_t i = 0; i < num_of_axes; ++i) {
+                if (signal_size[i] == -1) {
+                    continue;
+                }
+                output_shape[axes[i]] = DimType(signal_size[i]);
+            }
+        } else if (signal_size_shape.rank().is_static()) {
+            for (int64_t axis : axes) {
+                output_shape[axis] = ov::Dimension::dynamic();
+            }
+        }
+    } else if (input_shape.rank().is_static() && (axes_shape.rank().is_dynamic() || !axes_are_known)) {
+        const auto input_rank = input_shape.size();
+        for (int64_t i = 0; i < input_rank - 1; ++i) {
+            output_shape[i] = ov::Dimension::dynamic();
+        }
+    }
+}
+
+template <class T>
+void shape_infer(const ov::op::v7::DFT* op,
+                 const std::vector<T>& input_shapes,
+                 std::vector<T>& output_shapes,
+                 const std::map<size_t, std::shared_ptr<ngraph::runtime::HostTensor>>& constant_data = {}) {
+    NODE_VALIDATION_CHECK(op, (input_shapes.size() == 2 || input_shapes.size() == 3) && output_shapes.size() == 1);
+    fft_base_shape_infer(op, input_shapes, output_shapes, constant_data);
+}
+
+template <class T>
+void shape_infer(const ov::op::v7::IDFT* op,
+                 const std::vector<T>& input_shapes,
+                 std::vector<T>& output_shapes,
+                 const std::map<size_t, std::shared_ptr<ngraph::runtime::HostTensor>>& constant_data = {}) {
+    NODE_VALIDATION_CHECK(op, (input_shapes.size() == 2 || input_shapes.size() == 3) && output_shapes.size() == 1);
+    fft_base_shape_infer(op, input_shapes, output_shapes, constant_data);
+}
```

Comment:
Can we rename `fft_base_shape_infer` to `shape_infer` and call it directly?
In this case, we don't need to have separate `shape_infer` functions for DFT and IDFT

Reply:
Yes, changed accordingly.